fullstack-projects/url-shortener/devashmit-day8/backend/main.py

```
@@ -0,0 +1,55 @@
+
+@app.get("/analytics", response_model=list[AnalyticsItem])
+def analytics():
+    with database.get_conn() as c:
+        rows = c.execute("SELECT * FROM urls ORDER BY created_at DESC").fetchall()
+    return [AnalyticsItem(**dict(r)) for r in rows]
+
+@app.get("/analytics/{code}", response_model=AnalyticsItem)
+def analytics_for(code: str):
+    with database.get_conn() as c:
+        row = c.execute("SELECT * FROM urls WHERE code=?", (code,)).fetchone()
+    if not row: raise HTTPException(404, "Not found")
```

Ruff (0.15.15)

[error] Multiple statements on one line (colon)

(E701)